src/interfaceimpl.jl

```diff
@@ -83,21 +83,44 @@ checksquareblocks(A) = blockisequal(axes(A)...) || throw(DimensionMismatch("bloc
 
 for op in (:-, :+)
     @eval begin
-        function $op(A::BlockTridiagonal, λ::UniformScaling) 
+        function $op(A::BlockDiagonal, λ::UniformScaling)
+            checksquareblocks(A)
+            mortar(Diagonal(broadcast($op, A.blocks.diag, Ref(λ))))
+        end
+        function $op(λ::UniformScaling, A::BlockDiagonal)
+            checksquareblocks(A)
+            mortar(Diagonal(broadcast($op, Ref(λ), A.blocks.diag)))
+        end
+
+        function $op(A::BlockTridiagonal, λ::UniformScaling)
             checksquareblocks(A)
             mortar(Tridiagonal(A.blocks.dl, broadcast($op, A.blocks.d, Ref(λ)), A.blocks.du))
         end
-        function $op(λ::UniformScaling, A::BlockTridiagonal) 
+        function $op(λ::UniformScaling, A::BlockTridiagonal)
             checksquareblocks(A)
             mortar(Tridiagonal(broadcast($op,A.blocks.dl), broadcast($op, Ref(λ), A.blocks.d), broadcast($op,A.blocks.du)))
         end
-        function $op(A::BlockBidiagonal, λ::UniformScaling) 
+        function $op(A::BlockBidiagonal, λ::UniformScaling)
             checksquareblocks(A)
             mortar(Bidiagonal(broadcast($op, A.blocks.dv, Ref(λ)), A.blocks.ev, A.blocks.uplo))
         end
-        function $op(λ::UniformScaling, A::BlockBidiagonal) 
+        function $op(λ::UniformScaling, A::BlockBidiagonal)
             checksquareblocks(A)
             mortar(Bidiagonal(broadcast($op, Ref(λ), A.blocks.dv), broadcast($op,A.blocks.ev), A.blocks.uplo))
         end
     end
-end
\ No newline at end of file
+end
+
+function replace_in_print_matrix(A::BlockDiagonal, i::Integer, j::Integer, s::AbstractString)
```

Comment:
Are you sure these are needed? Since `BlockDiagonal <: BlockArray <: LayoutArray` these should automatically deduce the sparsity in printing via `colsupport`, as they lower to:

https://github.com/JuliaMatrices/ArrayLayouts.jl/blob/5e0f27556d101d005396725c244562772673d734/src/ArrayLayouts.jl#L233

Reply:
You're right. It could be that they were needed when I started the PR, but not anymore.